src/main/java/spireMapOverhaul/zones/manasurge/modifiers/AbstractManaSurgeModifier.java

```diff
@@ -0,0 +1,43 @@
+package spireMapOverhaul.zones.manasurge.modifiers;
+
+import basemod.abstracts.AbstractCardModifier;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+
+public abstract class AbstractManaSurgeModifier extends AbstractCardModifier {
+
+    public boolean isPermanent;
+
+    public enum ModRarity {
+        COMMON_MOD,
+        UNCOMMON_MOD
+    }
+
+    public enum ModEffect {
+        POSITIVE_MOD,
+        NEGATIVE_MOD
+    }
+
+    private final ModRarity modRarity;
+    private final ModEffect modEffect;
+
+
+    public AbstractManaSurgeModifier(boolean permanent, ModRarity modRarity, ModEffect modEffect) {
+        isPermanent = permanent;
+        this.modRarity = modRarity;
+        this.modEffect = modEffect;
+    }
+
+    public ModRarity getModRarity() {
+        return modRarity;
+    }
+
+    public ModEffect getModEffect() {
+        return modEffect;
+    }
+
+
+    @Override
+    public boolean removeAtEndOfTurn(AbstractCard card) {
+        return !card.retain && !isPermanent;
```

Comment:
card.selfRetain also exists. retain is for temporary effects like well laid plans

Reply:
Oh, I thought retain contained both self and regular retain. so it should be return `!card.selfRetain && !card.retain && !isPermanent;` instead?

Reply:
Done either way.